### PR TITLE
DOCS-4924 Add Integration Assets Reference Shortcode

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Migrated the `manifest.json` file assets from [Integration Assets Reference](https://docs.datadoghq.com/developers/integrations/check_references/) into a shortcode to work around the table formatting + generates a preview for https://github.com/DataDog/integrations-core/pull/14163 (merge in the integrations-core PR before this PR).

### Motivation

DOCS-4924

### Preview

https://docs-staging.datadoghq.com/alai97/integrations-assets-reference-categories-update/developers/integrations/check_references/

### Additional Notes

Remember to change back the `pull_config_preview.yaml` and the `cache_enabled` flag before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 1]
```diff
@@ -86,14 +86,14 @@
           - "*/datadog_checks/*/__about__.py"
 
         - action: pull-and-push-folder
-          branch: master
+          branch: alai97/integration-assets-reference-doc-update
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:89]:
```suggestion
          branch: master
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -86,14 +86,14 @@
...
           globs:
           - docs/dev/*.md
           options:
             dest_dir: '/developers/integrations/'
             path_to_remove: 'docs/dev/'
             front_matters:
-              dependencies: [ "https://github.com/DataDog/integrations-core/blob/master/docs/dev/" ]
+              dependencies: [ "https://github.com/DataDog/integrations-core/blob/alai97/integration-assets-reference-doc-update/docs/dev/" ]
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:96]:
```suggestion
              dependencies: [ "https://github.com/DataDog/integrations-core/blob/master/docs/dev/" ]
```

**Review by alai97 — COMMENTED** [review]
To commit once https://github.com/DataDog/integrations-core/pull/14163 is merged.

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 - config:
-    cache_enabled: true
+    cache_enabled: false
```
Comment (alai97) [inline local/bin/py/build/configurations/pull_config_preview.yaml:3]:
```suggestion
    cache_enabled: true
```

File: layouts/shortcodes/integration-assets-reference.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+You can find the complete list of mandatory and optional attributes for the `manifest.json` file below: 
+
+| Attribute                                            | Mandatory or Optional       | Description                                                                                                                                                                                                                                                                       |
+|------------------------------------------------------|-----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (hestonhoffman) [inline layouts/shortcodes/integration-assets-reference.md:4]:
```suggestion
|----------|----------|----------|
```
Shortening these just makes it a little easier to parse in PRs, IDEs etc

**Review by hestonhoffman — APPROVED** [review]
This LGTM - I think there are probably a couple more changes needed in the `integrations.py`   file. Websites will want to take a look there I think.

**Review by nmuesch — APPROVED** [review]
Aside from your inline comments about switching the build flags back to normal (`master`, the cache setting), this LGTM! The preview looks great 🙇 🚀

**Review by hestonhoffman — APPROVED** [review]

**Review by anawishnoff — APPROVED** [review]

**Pull request merged**